src/audio/base_fw_intel.c

```diff
@@ -314,3 +315,52 @@ int basefw_vendor_set_large_config(struct comp_dev *dev,
 
 	return IPC4_UNKNOWN_MESSAGE_TYPE;
 }
+
+static inline bool is_ssp_node_id(uint32_t dma_type)
+{
+	return dma_type == ipc4_i2s_link_output_class ||
+	       dma_type == ipc4_i2s_link_input_class;
+}
+
+int basefw_vendor_dma_control(uint32_t node_id, const char *config_data, size_t data_size)
+{
+	union ipc4_connector_node_id node = (union ipc4_connector_node_id)node_id;
+	int ret, result;
+
+	tr_info(&basefw_comp_tr, "node_id 0x%x, config_data 0x%x, data_size %u",
+		node_id, (uint32_t)config_data, data_size);
+	if (!is_ssp_node_id(node.f.dma_type)) {
```

Comment:
I think this check should be removed. UAOL and DMIC interfaces will also implement DMA_CONTROL IPC to update dai configuration in runtime.

Reply:
Thank you for pointing that out. The check for SSP-specific node IDs has been removed as the `dai_config_update` function will return `-ENOSYS` if a DAI driver does not implement the required operation. This ensures that the `IPC4_DMA_CONTROL` handling is appropriately generalized for all DAI types.